src/ergw_aaa_diameter.erl

```diff
@@ -373,24 +380,24 @@ create_ACR(Session, Type, State) ->
        'Tunneling' = tunneling(Session, State),
        '3GPP-IMSI' = attr_get('3GPP-IMSI', Session),
        '3GPP-Charging-Id' = attr_get('3GPP-Charging-ID', Session),
-       '3GPP-PDP-Type' = [pdp_type(attr_get('3GPP-Charging-ID', Session))],
+       '3GPP-PDP-Type' = [pdp_type(attr_get('3GPP-PDP-Type', Session))],
        '3GPP-CG-Address' = attr_get_addr('3GPP-Charging-Gateway-Address', Session),
        '3GPP-GPRS-Negotiated-QoS-Profile' = attr_get('3GPP-GPRS-Negotiated-QoS-Profile', Session),
        '3GPP-SGSN-Address' = attr_get_addr('3GPP-SGSN-Address', Session),
        '3GPP-GGSN-Address' = attr_get_addr('3GPP-GGSN-Address', Session),
        '3GPP-IMSI-MCC-MNC' = attr_get('3GPP-IMSI-MCC-MNC', Session),
        '3GPP-GGSN-MCC-MNC' = attr_get('3GPP-GGSN-MCC-MNC', Session),
-       '3GPP-NSAPI' = attr_get('3GPP-NSAPI', Session),
-       '3GPP-Selection-Mode' = attr_get('3GPP-Selection-Mode', Session),
+       '3GPP-NSAPI' = octet_string(attr_get('3GPP-NSAPI', Session)),
```

Comment:
this could be wrong, 3GPP TS 29.061, Section 16.4.7.2 specifies the encoding for RADIUS of NSAPI to be a hexadecimal string, the table in Section 16a.5 implies to me that the same encoding is to be used for DIAMETER.